tsz-macro/src/lib.rs

```diff
@@ -514,52 +514,168 @@ pub fn derive_compressv2(tokens: TokenStream) -> TokenStream {
+    let col_comp_queue_idents = col_idents
         .iter()
-        .map(|ident| format_ident!("{}_compressor", ident))
+        .map(|ident| format_ident!("{}_compressor_queue", ident))
         .collect_vec();
-    let col_output_idents = col_idents
+    let col_delta_buf_idents = col_idents
         .iter()
-        .map(|ident| format_ident!("{}_output", ident))
+        .map(|ident| format_ident!("{}_delta_output_buffer", ident))
+        .collect_vec();
+    let col_delta_delta_buf_idents = col_idents
+        .iter()
+        .map(|ident| format_ident!("{}_delta_delta_output_buffer", ident))
         .collect_vec();
     let num_columns = col_idents.len();
-
+    let col_values_emitted_delta = col_idents
+        .iter()
+        .map(|ident| format_ident!("{}_columns_values_emitted_delta_compression", ident))
+        .collect_vec();
+    let col_values_emitted_delta_delta = col_idents
+        .iter()
+        .map(|ident| format_ident!("{}_columns_values_emitted_delta_delta_compression", ident))
+        .collect_vec();
     let compressor_struct = quote! {
         struct #compressor_ident {
-            #( #col_comp_idents: ::tsz_compress::prelude::CompressionQueue<#col_tys, 10>,)*
-            #( #col_output_idents: ::tsz_compress::prelude::BitBuffer,)*
-            column_values_emitted: usize,
+            #( #col_comp_queue_idents: ::tsz_compress::prelude::CompressionQueue<#col_tys, 10>,)*
+            #( #col_delta_buf_idents: Option<::tsz_compress::prelude::BitBuffer>,)*
+            #( #col_delta_delta_buf_idents: Option<::tsz_compress::prelude::BitBuffer>,)*
+            #( #col_values_emitted_delta: usize,)*
+            #( #col_values_emitted_delta_delta: usize,)*
+
         }
 
         impl TszCompressV2 for #compressor_ident {
             type T = #ident;
 
+            /// Performs compression using either delta or delta-delta compression, selecting the method that yields the smallest compressed values.
             fn compress(&mut self, row: Self::T) {
+
+                use config::Config;
+                let mut settings = Config::default();
+                settings.merge(config::File::with_name("config.toml")).unwrap();
+                let COMPRESSION_SIZE_FACTOR: usize = settings.get("COMPRESSION_SIZE_FACTOR").unwrap_or(100);
+
                 #(
-                    self.#col_comp_idents.push(row.#col_idents);
-                    if self.#col_comp_idents.is_full() {
-                        self.column_values_emitted += emit_bits(&mut self.#col_comp_idents, &mut self.#col_output_idents, false);
+                    self.#col_comp_queue_idents.push(row.#col_idents);
+                    // Enqueues values until the queue reaches its maximum capacity.
+                    if self.#col_comp_queue_idents.is_full() {
+                        self.#col_delta_buf_idents.as_mut().map(|outbuf| self.#col_values_emitted_delta += self.#col_comp_queue_idents.emit_delta_bits( outbuf, false));
```

Comment:
The first two rows need to be emitted as SVLQs for full and delta before the run of encoded values that are dominantly delta or delta-delta.